src/engine/controls/cuecontrol.cpp

```diff
@@ -1800,17 +2062,30 @@ HotcueControl::HotcueControl(QString group, int i)
         : m_group(group),
           m_iHotcueNumber(i),
           m_pCue(NULL),
-          m_bPreviewing(false),
+          m_previewingType(mixxx::CueType::Invalid),
           m_previewingPosition(-1) {
     m_hotcuePosition = new ControlObject(keyForControl(i, "position"));
     connect(m_hotcuePosition, &ControlObject::valueChanged,
             this, &HotcueControl::slotHotcuePositionChanged,
             Qt::DirectConnection);
     m_hotcuePosition->set(Cue::kNoPosition);
 
+    m_hotcueLength = new ControlObject(keyForControl(i, "length"));
```

Comment:
What would be a use case for this?

Reply:
Oh I see it is used by WaveformMark. Maybe an "endposition" CO would be more straightforward to use.